webapp-django/crashstats/supersearch/static/supersearch/js/socorro/search.js

```diff
@@ -183,78 +213,82 @@ $(function () {
         func.call(window.history, params, 'Search results', url + hash);
     }
 
-    function sortResults(results, container, query) {
-        if (query.term) {
-            return results.sort(function (a, b) {
-                if (a.text.length > b.text.length) {
-                    return 1;
-                }
-                else if (a.text.length < b.text.length) {
-                    return -1;
-                }
-                else {
-                    return 0;
-                }
-            });
-        }
-        return results;
-    }
-
     /**
-     * Simple Search UI with a limited number of fields.
+     * Return the current parameters defined by the search form.
      */
-    var SimpleSearch = {
-        container: $('#simple-search'),
-        initialized: false
-    };
-
-    SimpleSearch.initialize = function () {
-        if (this.initialized) {
-            return;
-        }
-
-        $('input[type=text]', this.container).select2({
-            'width': 'element',
-            'tags': []
-        });
-        $('select', this.container).select2({
-            'width': 'element',
-            'closeOnSelect': false
-        });
-
-        this.initialized = true;
-    };
+    function getParams() {
+        var params = form.dynamicForm('getParams');
 
-    SimpleSearch.getParams = function () {
-        var params = {};
-        $('select', this.container).each(function (i, item) {
+        $('select', simpleSearchContainer).each(function (i, item) {
             var name = item.name;
             var value = $(item).select2('val');
             if (value.length) {
-                params[name] = value;
+                if (params[name]) {
+                    params[name] = params[name].concat(value);
+                }
+                else {
+                    params[name] = value;
+                }
             }
         });
+
         return params;
-    };
+    }
+
+    /**
+     * Update the search form with new parameters.
+     */
+    function setParams(params) {
+        function hasOperator(value) {
+            var operators = form.dynamicForm('getOperatorsList');
+            return operators.some(function (operator) {
+                if (operator === 'has') {
+                    return false; // we don't care about that one
+                }
+                return value.indexOf(operator) === 0;
+            });
+        }
```

Comment:
I didn't get the joke, but that function should probably be out of the first one. :)